sdk/python/kfp/dsl/types/artifact_types.py

```diff
@@ -89,7 +89,8 @@ def _get_path(self) -> Optional[str]:
             return _MINIO_LOCAL_MOUNT_PREFIX + self.uri[len('minio://'):]
         elif self.uri.startswith('s3://'):
             return _S3_LOCAL_MOUNT_PREFIX + self.uri[len('s3://'):]
-        return None
+        # uri == path for local execution
+        return self.uri
```

Comment:
@chensun, WDYT about this change?

It is a behavior change. If a user depends on `.path` being `None`, their code's behavior may change. This doesn't affect code paths that write to `.path` (since you cannot write to `None`), but it would affect code like this:

```python
@dsl.component
def my_comp(a: Artifact):
    if a.path is None: # prev evaluates to True, now would evaluate to False
        persist('some artifact contents', a.uri)
    else:
        persist('some artifact contents', a.path)
```

The alternative is to set some sort of special flag on both input and output artifacts to allow `.path` to return the "local uri" when executing locally:

```python
# implementation code
for k, maybe_artifact in user_kwargs.items():
    if running_locally and isinstance(maybe_artifact, dsl.Artifact):
        maybe_artifact._use_uri_as_path = True

# then change `.path` to respect the `._use_uri_as_path` flag
```

WDYT? The reason for not immediately proceeding with the alternative implementation above is that:
- The current implementation's behavior may be acceptable in a minor version bump with release notes.
- The solution above is somewhat hacky and it's challenging to exhaustively reason about whether it will play nicely with user local execution code or, eventually, input artifact constants.

I can go either way on this.

Reply:
> If a user depends on .path being None, their code's behavior may change.

This sounds fine to me. I don't perceive `.path` being `None` an existing scenario until very recent changes for local execution.